usb: device: Revert "Do not call callback when transfer is cancelled"

jfischer-no opened the pull request:
This reverts commit f206170c655c433f25691728a22d675e15ad6d78 introduced as workaround for nRF USBD device controller in PR https://github.com/zephyrproject-rtos/zephyr/pull/16193.

This commit may be reverted due to changes made in commit e326c5839938
("usb: device: Do not cancel transfers on suspend").

Cherry-picked from https://github.com/zephyrproject-rtos/zephyr/pull/53540 and fixed commit message.

NickolasLapp committed:
usb: device: Revert "Do not call callback when transfer is cancelled"

This reverts commit f206170c655c433f25691728a22d675e15ad6d78
introduced as workaround for nRF USBD device controller in PR
https://github.com/zephyrproject-rtos/zephyr/pull/16193.

This commit may be reverted due to changes made in
commit e326c5839938
("usb: device: Do not cancel transfers on suspend").

Signed-off-by: Nickolas Lapp <nickolaslapp@gmail.com>
Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

jfischer-no commented:
Perhaps we should also revert cd7461495f1aeb4561d6a7e96c9f37b029805694.

tmon-nordic commented:
> Perhaps we should also revert [cd74614](https://github.com/zephyrproject-rtos/zephyr/commit/cd7461495f1aeb4561d6a7e96c9f37b029805694).

Yes, seems that cd7461495f1aeb4561d6a7e96c9f37b029805694 was entirely to workaround f206170c655c433f25691728a22d675e15ad6d78

jfischer-no committed:
usb: device: Revert "fix possible deadlock in usb_transfer_sync()"

This reverts commit cd7461495f1aeb4561d6a7e96c9f37b029805694,
introduced to fix possible deadlock as result of changes in
commit f206170c655c
("usb: device: Do not call callback when transfer is cancelled")
which is now reverted.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

jfischer-no committed:
usb: add a note about the return value of usb_transfer_sync()

The transfer completion callback does not provide a way to pass
transfer status to the transfer submitter.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>
